PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx

```diff
@@ -0,0 +1,367 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file taskFlowCharmHadrons.cxx
+/// \brief Analysis task for charm hadron flow
+///
+/// \author S. Politanò, INFN Torino, Italy
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "Framework/runDataProcessing.h"
+
+#include "Common/Core/EventPlaneHelper.h"
+#include "Common/DataModel/Qvectors.h"
+
+#include "PWGHF/Core/HfHelper.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+
+using namespace o2;
+using namespace o2::aod;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+enum DecayChannel { DplusToPiKPi = 0,
+                    DsToKKPi,
+                    DsToPiKK };
+
+enum centralityEstimator { V0A = 0,
+                           T0M,
+                           T0A,
+                           T0C };
+
+enum qvecEstimator { FV0A = 0,
+                     FT0M,
+                     FT0A,
+                     FT0C,
+                     TPCPos,
+                     TPCNeg };
+
+struct HfTaskFlowCharmHadrons {
+  Configurable<int> harmonic{"harmonic", 2, "harmonic number"};
+  Configurable<int> qvecDetector{"qvecDetector", 0, "Detector for Q vector estimation (FV0A: 0, FT0M: 1, FT0A: 2, FT0C: 3, TPC Pos: 4, TPC Neg: 5)"};
+  Configurable<int> centDetector{"centDetector", 0, "Detector for centrality estimation (V0A: 0, T0M: 1, T0A: 2, T0C: 3"};
+  Configurable<int> selectionFlag{"selectionFlag", 1, "Selection Flag for hadron (e.g. 1 for skimming, 3 for topo. and kine., 7 for PID)"};
+  Configurable<bool> storeMl{"storeMl", false, "Flag to store ML scores (e.g. 0 for Bkg, 1 for Prompt, 2 for Feed-down)"};
+  Configurable<bool> saveEpResoHisto{"saveEpResoHisto", false, "Flag to save event plane resolution histogram"};
+  Configurable<std::vector<int>> classMl{"classMl", {0, 2}, "Indexes of BDT scores to be stored (e.g. 0 for Bkg, 2 for Feed-down). Two indexes max."};
```

Comment:
The example you make is not correct in general, the order depends on how the model is trained.